lms/sql_tasks/tasks/report/create_from_scratch/01_functions/01_date_functions.sql

```diff
@@ -23,6 +23,28 @@ AS $$
 $$
 LANGUAGE SQL;
 
+-- A function for getting intervals. This accepts the same resolutions as
+-- `report.multi_truncate` above.
+
+CREATE OR REPLACE FUNCTION report.single_interval(resolution TEXT)
+RETURNS INTERVAL
+IMMUTABLE
+AS $$
+    SELECT CASE
+        -- Our start date for "all time" is '1901' so add 200 years to ensure
+        -- the other end is far in the future
+        WHEN resolution = 'all_time' THEN INTERVAL '200 years'
```

Comment:
The Y0.2K problem

Reply:
Yeah. This is really going to bite us in the butt in 79 years.